### PR TITLE
Fix #4: Ability to provide options

### LINKED ISSUE #4: Add default options
As with [kenzanlabs/pipeline-validate-css](https://github.com/kenzanlabs/pipeline-validate-css) and [kenzanlabs/pipeline-validate-js](https://github.com/kenzanlabs/pipeline-validate-js), a set of preferred options should be included in this repo. 

These "default" options should have the ability to be overwritten by user provided config.

### PR DESCRIPTION
**Pull request by paul-barry-kenzan**
**Associated Issue**
#4 

**Changes Made**
- Add a set of default options via a `.htmllintrc` file
- Add ability for a custom `.htmllintrc` file (provided rules overwrite default)
- Add ability for a custom config object to be provided
- When both a file path AND a config object are provided, they are merged in a way that the custom rules object takes priority over the custom file and default rules
- Update README with usage details

### CODE REVIEW

File: .htmllintrc [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+    "attr-bans": [
```
Comment (thescientist13) [inline .htmllintrc]: indentation seems funky here in a couple of places

File: package.json [hunk 2]
```diff
@@ -48,14 +48,15 @@
...
     "pipeline-handyman": "1.0.0"
   },
   "devDependencies": {
     "chai": "3.3.0",
     "dirty-chai": "1.2.2",
     "gulp": "3.9.1",
+    "gulp-util": "3.0.8",
     "isstream": "0.1.2",
     "pipeline-test-node": "1.0.1",
     "pipeline-validate-js": "1.0.1",
```
Comment (thescientist13) [inline package.json]: bump here if we can

**Comment by thescientist13** [comment]
@paul-barry-kenzan 
Not necessarily for this PR, and can certainly be offloaded to a new issue, but what is the support for partial templates with this pipeline?  The fixtures I see are (relatively) complete HTML documents (eg. `<html>`, `<head>`, `<body>` tags) etc.  But in the case of a framework like Angular, it would be very common to have partial templates.

Thoughts?

**Comment by paul-barry-kenzan** [comment]
@thescientist13 
That's a great point. l [created an issue to track the effort](https://github.com/kenzanlabs/pipeline-validate-html/issues/7) to ensure that this pipeline works with partials.

**Pull request merged**